flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowDeserializationSchema.java

```diff
@@ -429,13 +428,21 @@ private Object convertField(
 		}
 	}
 
-	private DeserializationRuntimeConverter assembleArrayConverter(DeserializationRuntimeConverter elementConverter) {
+	private DeserializationRuntimeConverter assembleArrayConverter(
+		TypeInformation<?> elementType,
+		DeserializationRuntimeConverter elementConverter) {
+
+		final Class<?> elementClass = elementType.getTypeClass();
+
 		return (mapper, jsonNode) -> {
-			ArrayNode node = (ArrayNode) jsonNode;
+			final ArrayNode node = (ArrayNode) jsonNode;
+			final Object[] array = (Object[]) Array.newInstance(elementClass, node.size());
```

Comment:
Just new a normal `Object[]` is enough?

Reply:
`new Object[node.size()]` and `(Object[]) Array.newInstance(elementClass, node.size())` are a bit different though it may not affect the result here because we have `elementConverter` to care about the element type. I'm just trying to make the expression's semantics more precise here :).